analyzer/codechecker_analyzer/analyzers/clangsa/analyzer.py

```diff
@@ -101,11 +101,72 @@ class ClangSA(analyzer_base.SourceAnalyzer):
     """
     ANALYZER_NAME = 'clangsa'
 
+    __ctu_autodetection = None
+
     def __init__(self, cfg_handler, buildaction):
         super(ClangSA, self).__init__(cfg_handler, buildaction)
         self.__disable_ctu = False
         self.__checker_configs = []
 
+    @classmethod
+    def analyzer_binary(cls):
+        return analyzer_context.get_context() \
+            .analyzer_binaries[cls.ANALYZER_NAME]
+
+    @classmethod
+    def analyzer_plugins(cls):
+        plugin_dir = analyzer_context.get_context().checker_plugin
+
+        clangsa_plugin_dir = env.get_clangsa_plugin_dir()
+        is_analyzer_from_path = env.is_analyzer_from_path()
+        if is_analyzer_from_path:
+            if not clangsa_plugin_dir:
+                return []
+
+            # If the CC_ANALYZERS_FROM_PATH and CC_CLANGSA_PLUGIN_DIR
+            # environment variables are set we will use this value as the
+            # plugin directory.
+            plugin_dir = clangsa_plugin_dir
+
+        if not plugin_dir or not os.path.exists(plugin_dir):
+            return []
+
+        return [os.path.join(plugin_dir, f)
+                for f in os.listdir(plugin_dir)
+                if os.path.isfile(os.path.join(plugin_dir, f))
+                and f.endswith(".so")]
+
+    @classmethod
+    def get_version(cls, env=None):
+        """ Get analyzer version information. """
+        version = [cls.analyzer_binary(), '--version']
+        try:
+            output = subprocess.check_output(version,
+                                             env=env,
+                                             universal_newlines=True,
+                                             encoding="utf-8",
+                                             errors="ignore")
+            return output
+        except (subprocess.CalledProcessError, OSError) as oerr:
+            LOG.warning("Failed to get analyzer version: %s",
+                        ' '.join(version))
+            LOG.warning(oerr)
+
+        return None
+
+    @classmethod
+    def version_info(cls):
```

Comment:
They are not strictly redundant, because `get_version()` returns the output of `clang --version` command as a string so it can be forwarded through `CodeChecker analyzers --details`. On the other hand `version_info()` returns a `ClangVersionInfo` object so the major and minor version numbers can be used as integers. I think these two could be merged somehow, because they really are functionally redundant. This merged version will be the part of the common analyzer interface, so this issue will be solved there.